chore: update spark docs (#268)
OpenLineage/docs: docs/integrations/spark/spark.md

----------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------
| spark.openlineage.transport.type                   | The transport type used for event emit, default type is `console`                                                                                   | http                                |
| spark.openlineage.namespace                        | The default namespace to be applied for any jobs submitted                                                                                          | MyNamespace                         |
| spark.openlineage.parentJobName                    | The job name to be used for the parent job facet. If defined, `spark.openlineage.parentRunId` is also required.                                     | ParentJobName                       |

> | spark.openlineage.parentRunId                      | The RunId of the parent job that initiated this Spark job. Needs to be valid 128-bit UUID.                                                          | xxxx-xxxx-xxxx-xxxx                 |

Reviewer: ```suggestion
| spark.openlineage.parentRunId                      | The RunId of the parent job that initiated this Spark job. Must be a valid 128-bit UUID.                                                          | xxxx-xxxx-xxxx-xxxx                 |
```

      "spark.jars.packages": "io.openlineage:openlineage-spark:1.0.0+",
      "spark.openlineage.transport.url": f"{openlineage_url}/api/v1/namespaces/{openlineage_namespace}/jobs/dump_orders_to_gcs/runs/{{{{lineage_run_id(run_id, task)}}}}?api_key={api_key}",
      "spark.openlineage.namespace": openlineage_namespace,

>       "spark.openlineage.parentJobName": dag.dag_id,

Reviewer: I would argue here. Parent job is actually a task of which parent job is DAG. That is also what `lineage_run_id` macro matches (task instance run id, not dag run run id).

Author: my thinking here is:
- in **airflow open lineage integration** open lineage events can be emitted for every task (either through defining inlets/outlets of collected automatically for some compatible tasks) produces open lineage events with dag.dag_id as parentJobName
- this is an example of running spark job from airflow but producing open lineage events from spark open lineage integration (instead of airflow open lineage integration)
- we should maintain consistent approach for what we consider a parent

and my opinion is:
- airflow task is too granular/low level to be treated as a parent and I think it is handled properly in airflow integration that dag.dag_id is a parent for any task
- in majority of metadata systems pipeline/dag is first-class citizen, not pipeline task
- maintaining consistent approach can give us more benefits as we can have same context for metadata/lineage information produced by different integrations

so few questions:
- why do you think task is a better parent than dag?
- what do we gain by this inconsistency?

Reviewer 2: @mgorsk1 my thinking is that our Airflow job names are concatenation of dag_id and task_id. If some metadata system can show full name, you'd already see the dag. 

Also if you want to have less granular distinction as a consumer, you can already do that by utilizing that information, while people who want to have more granular one can use full job name. Also, we want to be able to have deeper hierarchies that single-level one, as in this case, dag -> task -> something spawned by task. So there are two ways to link spawned job to dag directly, and "collapse" them on UI.

>why do you think task is a better parent than dag?

I think it better models the reality. We do not prescribe how data should be displayed or linked (with some exceptions - consistent dataset naming) by different metadata systems, but want to have standard and integrations that is closest to actual execution, so that authors of that systems can trust that information to build their own vision of a graph.

Author: hej @mobuchowski indeed for Airflow `job.name` is `dag_id` + `task_id` and I don't mind it, my thoughts are more focused on `parent` facet for spark listener integration. 

Currently, in lineage events produced by airflow integration, if a task `process_data` from `sample_dag` DAG produces lineage information, we would have:
- `job.name = 'sample_dag-process_data'`
- `run.parent.job.name = 'sample_dag'`

and I think it describes reality quite well.

At the same time, when considering spark lineage events, `job.name` is more granular (but I don't really have a problem with it) as it is a concatenation of spark.appName and spark stage title. For example:

```
"job": {
      "namespace": "default",
      "name": "mysparkappname.execute_create_data_source_table_as_select_command.x_test_select_from_y",
      "facets": {}
    }
```

(`mysparkappname` is `spark.appName` while everything else is stage information)

My PR addressed the area where there is a GAP what do we understand by parent job for spark app. My initial thoughts were to make it  consistent with airflow approach but you brought up good point about OL events priority to depict actual execution reality, not enforce consistency. With that in mind, I think that:
- it's not totally ok that spark lineage events don't have default `run.parent.job.name`.
- if spark `job.name` contains `spark.appName`, we should default `run.parent.job.name` to this value - then we wouldn't need to provide `spark.openlineage.parentJobName` at all and it would be as you intend it - as close to actual execution as possible.

This I think would be even better, as, when thinking about actual execution:

- openlineage event from airflow is should have `job.name` inheriting from `task_id` while `run.parent.job.name` should be `dag_id`. It's a very accurate description of actual state and it's already implemented like that.
- openlineage event from spark job should have `job.name` to be a concatenation of spark.appName with spark stage name (this is already implemented like that) while `run.parent.job.name` should default to `spark.appName` and it would, as well, depict realiy quite precisely. 

 With that, we can withdraw this PR in favor of removing `spark.openlineage.parentJobName` from example completely and providing default parent run name in spark integration. What do you think?
